src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1384,77 +1382,66 @@ Literal &Literal::operator+=(const Literal &other) noexcept {
 }
 
 Literal Literal::sub(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
-    // TODO subtype handling
-
     // DateTime(&Subclasses) - DateTime(&Subclasses)
-    if (this->is_timepoint_like() && other.is_timepoint_like() && this->datatype_id() == other.datatype_id()) {
-        auto this_dt = this->cast_to_value<datatypes::xsd::DateTime>();
-        if (!this_dt.has_value())
-            return Literal{};
-
-        auto other_dt = other.cast_to_value<datatypes::xsd::DateTime>();
-        if (!other_dt.has_value())
-            return Literal{};
-
-        datatypes::registry::util::CheckedZonedTime const this_tp{this_dt->second.has_value() ? *this_dt->second : util::Timezone{},
-                                                                  datatypes::registry::util::to_checked(this_dt->first)};
-
-        datatypes::registry::util::CheckedZonedTime const other_tp{other_dt->second.has_value() ? *other_dt->second : util::Timezone{},
-                                                                   datatypes::registry::util::to_checked(other_dt->first)};
-
-        auto d = this_tp.get_sys_time() - other_tp.get_sys_time();
-        if (d.count().is_invalid())
-            return Literal{};
+    if (this->datatype_id() == other.datatype_id()) {
```

Comment:
Does the spec specify that you can only do `sub` if the types are equal?

Reply:
It specifies a function for each datatype with both parameters of the same type.
(Also, does `Time - Date` make any sense? Typecasting (as described in xsd) makes it `1972 - Date` which is likely not what the average user expects)

Reply:
No the `Time - Date` thing doesn't make much sense. But I would say subtracting two different date-like types does make sense. But if the spec says there is only a function for equal types then thats fine